DeviceConnectionAndReliabilityReadme.md

```diff
@@ -0,0 +1,62 @@
+    - SasTokenTimeToLive : The suggested time to live value for tokens generated for SAS authenticated clients. Default value is 60 minutes.
+    - SasTokenRenewalBuffer : The time buffer before expiry when the token should be renewed, expressed as a percentage of the time to live. Acceptable values lie between 0 and 100.Default value is 15%.
+
+    Note: If the shared access policy name is not specified in the connection string, it will be set to - `<iotHubHostName>/devices/<deviceId>`
+>  
+- Providing only the shared access signature
+
+    If you only provide the shared access signature and authenticationMethod is not of type AuthenticationWithTokenRefresh, there will be no automatic renewal.  
+>  
+- Providing your own SAS token using [DeviceAuthenticationWithTokenRefresh](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/DeviceAuthenticationWithTokenRefresh.cs) authenticationMethod
+
+    If you choose to use your own implementation to generate tokens, you can provide the time to live and time buffer before exipry through the DeviceAuthenticationWithTokenRefresh constructor. The ClientOptions apply only to other authentication methods.
+
+When using x509 certificates, [DeviceAuthenticationWithX509Certificate](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/DeviceAuthenticationWithX509Certificate.cs) authenticationMethod can be used. The client authentication will be valid until the certificate is valid. Any renewal will have to be done manually and the client needs to be recreated.
+
+### Connection retry logic
+
+For both AMQP and MQTT, the SDK will try to re-connect everytime there is any disruption. The default retry policy does not have a time limit and will follow exponential back-off.
+
+For more details, see [here](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/devdoc/retrypolicy.md).
+
+HTTP is a stateless protocol and will work whenever there is network connectivity.
+
+### Timeout controls
+
+There are different timeout values that can be configured for the DeviceClient based on the protocol. These values are configuarable through the following transport settings that are passed while creating the client. There will be no changes to any of the settings once the DeviceClient is created. The client needs to be recreated with new settings to make changes.
+
+AMQP transport settings - [AmqpTransportSettings](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/AmqpTransportSettings.cs)
+
+* IdleTimeout - The interval, in seconds, that the client establishes with the service, for sending keep alive pings. The default value is 2 minutes.
+* OperationTimeout - The time to wait for any operation to complete. The default is 1 minute.
+* OpenTimeout - This value is not used (TODO: Confirm and update)
```

Comment:
When I looked at the implementation, it seemed like we use OperationTimeout for Open call as well. Let me know if this is not correct. Otherwise it would be worth updating that this value is unused by our SDK.

Reply:
@abhipsaMisra - wanted to check on this